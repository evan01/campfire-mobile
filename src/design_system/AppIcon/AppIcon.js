// @flow
import React from "react";
import Icon from "react-native-vector-icons/dist/MaterialIcons";
import PropTypes from "prop-types";

const defaultProps = {
  name: "cake",
  size: 60,
  color: "black",
};

const propTypes = {
  name: PropTypes.string,
  size: PropTypes.number,
  color: PropTypes.string,
};

const AppIcon = ({ name, size, color }) => (
  <Icon name={name} size={size} color={color} />
);

AppIcon.propTypes = propTypes;
AppIcon.defaultProps = defaultProps;
export default AppIcon;
