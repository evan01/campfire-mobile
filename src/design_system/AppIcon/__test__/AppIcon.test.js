// @flow
import React from "react";
import renderer from "react-test-renderer";
import AppIcon from "../AppIcon";

describe("AppIcon", () => {
  test("renders correctly", () => {
    const Icon = renderer.create(<AppIcon name="bug-report" />);
    expect(Icon).toMatchSnapshot();
  });

  test("renders with different color", () => {
    const Icon = renderer.create(
      <AppIcon name={"star"} size={30} color={"orange"} />
    );
    expect(Icon).toMatchSnapshot();
  });
});
