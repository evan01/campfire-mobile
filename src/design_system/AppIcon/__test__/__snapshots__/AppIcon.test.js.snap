// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon renders correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontSize": 60,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`AppIcon renders with different color 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "orange",
        "fontSize": 30,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
